File: javascript/7kyu/uniqueStringCharacters.js
// You will be given two strings a and b and your task will be to return
// the characters that are not common in the two strings.

// For example:

// solve("xyab","xzca") = "ybzc" 
// --The first string has 'yb' which is not in the second string. 
// --The second string has 'zc' which is not in the first string. 

// Notice also that you return the characters from the first string
// concatenated with those from the second string.

function solve(...strings){
  const result = []
  strings.forEach((string, index) => {
    const notInS2 = [];
    const notInS1 = [];

    if (index === 0) {
      string.split('').forEach(char => {
        if (!strings[1].includes(char)) {
          notInS2.push(char)
        }
      })
    }
    
    if (index === 1) {
      string.split('').forEach(char => {
        if (!strings[0].includes(char)) {
          notInS1.push(char)
        }
      })
    }

    result.push(notInS2, notInS1)
  })

  return result.join().replace(/,/g, '');
}

console.log(solve("xyab","xzca"));
// "ybzc"

File: ruby/7kyu/unique_string_characters.rb
# In this Kata, you will be given two strings a and b and your task will be
# to return the characters that are not common in the two strings.

# For example:

# solve("xyab","xzca") = "ybzc" 
# --The first string has 'yb' which is not in the second string. 
# --The second string has 'zc' which is not in the first string. 

# Notice also that you return the characters from the first string
# concatenated with those from the second string.

# initial solution
def solve(a,b)
  not_common = []
  a.split('').each do |char|
    if !b.include?(char)
      not_common << char
    end
  end

  b.split('').each do |char|
    if !a.include?(char)
      not_common << char
    end
  end

  not_common.join
end

p solve("xyab","xzca")
# "ybzc" 